runtime/vm/createramclass.cpp

```diff
@@ -2405,7 +2405,7 @@ internalCreateRAMClassFromROMClassImpl(J9VMThread *vmThread, J9ClassLoader *clas
 #if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
 			allocationRequests[RAM_CLASS_FLATTENED_CLASS_CACHE].prefixSize = 0;
 			allocationRequests[RAM_CLASS_FLATTENED_CLASS_CACHE].alignment = OMR_MAX(sizeof(J9Class *), sizeof(UDATA));
-			allocationRequests[RAM_CLASS_FLATTENED_CLASS_CACHE].alignedSize = sizeof(J9FlattenedClassCache) * (flattenedClassCache[0].offset > 0 ? flattenedClassCache[0].offset + 1 : 0);
+			allocationRequests[RAM_CLASS_FLATTENED_CLASS_CACHE].alignedSize = sizeof(J9FlattenedClassCache) * (flattenedClassCache[0].offset + 1);
```

Comment:
Around line 2927 add comment that describes the structure of the flattened class cache

e.g.

--------------------------------
|  default Value | empty | size    |
--------------------------------
|         clazz1            | NaS1   | offset1|
--------------------------------
|         clazzN            | NaSN | offsetN|
--------------------------------

Perhaps an ascii picture